components/camel-hazelcast/src/main/java/org/apache/camel/processor/aggregate/hazelcast/HazelcastAggregationRepository.java

```diff
@@ -290,6 +290,15 @@ public Exchange get(CamelContext camelContext, String key) {
         return unmarshallExchange(camelContext, cache.get(key));
     }
     
+    /**
+     * Checks if the key in question is in the repository.
+     * 
+     * @param key Object - key in question
+     */
+    public boolean containsKey(Object key){
+        return cache.containsKey(key);
```

Comment:
Not quite sure about it. The persistedCache is used only in the recoveryMode. As far as i've seen entries are added to the persistedCache only in the remove method after the entry has been removed from cache. So i think we should only look into the cache.

Also the existing getKeys method does only look into the cache too but only retrieves the keys from the cache.